File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uset; 
        for(auto& i:nums) uset.insert(i); 
        int res = 0,cnt=0;
        for(auto& i:uset){
            if(uset.find(i-1)==uset.end()) {
                int val = i+1;
                cnt=1;
                while(uset.find(val)!=uset.end()){
                    val++;
                    cnt++;  
                }
                res=max(res,cnt);
            }
            else continue;
        }
        return res;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_nums = set(nums)
        if len(set_nums)==0: 
            return 0
        cnt,res=1,1
        for x in set_nums: 
            val = x
            cnt = 0 
            if (val-1) in set_nums: 
                continue 
            else: 
                while val in set_nums:
                    cnt+=1 
                    # set_nums.remove(val) 
                    val+=1 
            res=max([res,cnt])
        return res